src/compute/src/sink/persist_sink.rs

```diff
@@ -881,234 +851,232 @@ where
         vec![Antichain::new()],
     );
 
-    // Dropping this token signals that the operator should shut down cleanly.
-    let token = Rc::new(());
-    let token_weak = Rc::downgrade(&token);
-
     // This operator accepts the batch descriptions and tokens that represent
     // written batches. Written batches get appended to persist when we learn
     // from our input frontiers that we have seen all batches for a given batch
     // description.
 
-    append_op.build_async(
-        scope,
-        move |mut capabilities, frontiers, scheduler| async move {
-            let mut cap_set = if active_worker {
-                CapabilitySet::from_elem(capabilities.pop().expect("missing capability"))
-            } else {
-                // Pop away, because we don't need it.
-                capabilities.pop().expect("missing capability");
-                CapabilitySet::new()
-            };
-
-            let mut description_buffer = Vec::new();
-            let mut batch_buffer = Vec::new();
-
-            // Contains descriptions of batches for which we know that we can
-            // write data. We got these from the "centralized" operator that
-            // determines batch descriptions for all writers.
-            let mut in_flight_descriptions: HashSet<(Antichain<Timestamp>, Antichain<Timestamp>)> =
-                HashSet::new();
-
-            let mut in_flight_batches: HashMap<
-                (Antichain<Timestamp>, Antichain<Timestamp>),
-                Vec<Batch<_, _, _, _>>,
-            > = HashMap::new();
-
-            // TODO(aljoscha): We need to figure out what to do with error results from these calls.
-            let persist_client = persist_clients
-                .lock()
-                .await
-                .open(persist_location)
-                .await
-                .expect("could not open persist client");
-
-            let mut write = persist_client
-                .open_writer::<SourceData, (), Timestamp, Diff>(
-                    shard_id,
-                    &format!("persist_sink::append_batches {}", sink_id),
-                )
-                .await
-                .expect("could not open persist shard");
-
-            while scheduler.notified().await {
-                if token_weak.upgrade().is_none() {
-                    return;
-                }
+    let shutdown_button = append_op.build(move |mut capabilities| async move {
+        if !active_worker {
+            // SUBTLE: This is different from `mint_batch_description`
+            // where the non-active workers have to stay alive and keep
+            // pumping input, to ensure that the one active worker sees
```

Comment:
This comment is obsolete now. Nothing needs to stay alive to pump input as this is happening automatically so we can remove this comment

Reply:
Ah, that means we can also make `mint_batch_description` return early on inactive workers! I'll make that part of a separate PR, this one is already large enough.